src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -0,0 +1,1116 @@
+        {
+            Assert.Equal(OperationKind.Using, operation.Kind);
+            AssertEx.Equal(new[] { operation.Resources, operation.Body }, operation.Children);
+        }
+
+        // https://github.com/dotnet/roslyn/issues/21281
+        internal override void VisitFixed(IFixedOperation operation)
+        {
+            Assert.Equal(OperationKind.None, operation.Kind);
+            AssertEx.Equal(new[] { operation.Variables, operation.Body }, operation.Children);
+        }
+
+        public override void VisitExpressionStatement(IExpressionStatementOperation operation)
+        {
+            Assert.Equal(OperationKind.ExpressionStatement, operation.Kind);
+            Assert.Same(operation.Operation, operation.Children.Single());
+        }
+
+        internal override void VisitWith(IWithOperation operation)
+        {
+            Assert.Equal(OperationKind.None, operation.Kind);
+            AssertEx.Equal(new[] { operation.Value, operation.Body }, operation.Children);
+        }
+
+        public override void VisitStop(IStopOperation operation)
+        {
+            Assert.Equal(OperationKind.Stop, operation.Kind);
+            Assert.Empty(operation.Children);
+        }
+
+        public override void VisitEnd(IEndOperation operation)
+        {
+            Assert.Equal(OperationKind.End, operation.Kind);
+            Assert.Empty(operation.Children);
+        }
+
+        public override void VisitInvocation(IInvocationOperation operation)
+        {
+            Assert.Equal(OperationKind.Invocation, operation.Kind);
+            Assert.NotNull(operation.TargetMethod);
+            var isVirtual = operation.IsVirtual;
+
+            IEnumerable<IOperation> children;
+            if (operation.Instance != null)
+            {
+                children = new[] { operation.Instance }.Concat(operation.Arguments);
+            }
+            else
+            {
+                children = operation.Arguments;
+            }
+
+            AssertEx.Equal(children, operation.Children);
+
+            // Make sure that all static member references or invocations of static methods do not have implicit IInstanceReferenceOperations
+            // as their receivers
+            if (operation.TargetMethod.IsStatic &&
+                operation.Instance is IInstanceReferenceOperation)
+            {
+                Assert.False(operation.Instance.IsImplicit, $"Implicit {nameof(IInstanceReferenceOperation)} on {operation.Syntax}");
+            }
+        }
+
+        public override void VisitArgument(IArgumentOperation operation)
+        {
+            Assert.Equal(OperationKind.Argument, operation.Kind);
+            Assert.Contains(operation.ArgumentKind, new[] { ArgumentKind.DefaultValue, ArgumentKind.Explicit, ArgumentKind.ParamArray });
+            var parameter = operation.Parameter;
+
+            Assert.Same(operation.Value, operation.Children.Single());
+            var inConversion = operation.InConversion;
```

Comment:
Should also test the extension methods to get the original conversions, and make sure that the opposite-language ones throw exceptions appropriately.

Reply:
> Should also test the extension methods to get the original conversions, and make sure that the opposite-language ones throw exceptions appropriately.

This is outside of scope of this PR. Feel free to open an issue to add additional validations.

---
In reply to: [157597636](https://github.com/dotnet/roslyn/pull/23840#discussion_r157597636) [](ancestors = 157597636)

Reply:
https://github.com/dotnet/roslyn/issues/23854